src/components/SelectPlayStep.tsx

```diff
@@ -8,8 +8,36 @@ interface Props {
 }
 
 export default class SelectPlayStep extends React.PureComponent<Props> {
+
+  renderChooseButton(choosePlay: (play: Play) => void, play: Play, description: string) {
+    return (
+      <button
+      type="button"
+      onClick={() => choosePlay(play)}
+      style={{ display: 'inline-block', width: '33%' }}
+      key={play}
+    >
+      <div style={{ height: 250, background: 'maroon', margin: 4 }}>
+        <div
+          style={{
+            left: '50%',
+            position: 'relative',
+            top: '50%',
+            transform: 'translate(-50%, -50%)',
+            width: 'fit-content',
+          }}
+        >
+          <h1>{description}</h1>
+        </div>
+      </div>
+    </button>
+    )
+  }
+
+
   render() {
-    const { choosePlay, afterOpponent } = this.props;
+    const { afterOpponent, choosePlay } = this.props;
+    const renderChooseButton = this.renderChooseButton;
```

Comment:
Just FYI this is why we couldn't access `this.props` in `renderChooseButton`.  It looks like when we do  `const renderChooseButton = this.renderChooseButton;` When `renderChooseButton`  executs `this` is the global context. See https://www.typescriptlang.org/play/#src=class%20Dummy%20%7B%0D%0A%20%20%20%20doStuff()%20%7B%0D%0A%20%20%20%20%20%20%20%20console.log(this)%3B%0D%0A%20%20%20%20%20%20%20%20console.log(this.data)%3B%0D%0A%20%20%20%20%7D%0D%0A%20%20%20%20%20doOtherStuff()%20%7B%0D%0A%20%20%20%20%20%20%20%20%20const%20doStuff%20%3D%20this.doStuff%3B%0D%0A%20%20%20%20%20%20%20%20%20this.doStuff()%3B%0D%0A%20%20%20%20%20%20%20%20%20doStuff()%3B%0D%0A%20%20%20%20%7D%0D%0A%0D%0A%20%20%20%20data%20%3D%205%3B%0D%0A%20%20%20%20%0D%0A%7D%0D%0A%0D%0Aconst%20dummy%20%3D%20new%20Dummy()%3B%0D%0Adummy.doOtherStuff()%3B

We can switch `   { renderChooseButton(choosePlay, Play.Rock, "Rock") }` to    `{this. renderChooseButton(Play.Rock, "Rock") }`. That way in `renderChooseButton` `this` is the React Component.